### ISSUE TITLE
Add .pre-commit-config.yaml and use pre-commit to run the various linters, formatters, and checks

### ISSUE THREAD

**Issue by yelircaasi** [LibreLingo/LibreLingo#3159]
[pre-commit](https://pre-commit.com/) is a tremendous project that has greatly improved my workflow - easy to install and easy to use. I would be happy to implement it here and open a pull request, I just wanted to see first if that is something people would be on board with. Its benefits are twofold - firstly, it ensures code standards are met before each commit. Secondly, it provides an easy command to run all relevant linters etc. at any point during the development process.

**Comment by kantord** [comment]
> [pre-commit](https://pre-commit.com/) is a tremendous project that has greatly improved my workflow - easy to install and easy to use. I would be happy to implement it here and open a pull request, I just wanted to see first if that is something people would be on board with. Its benefits are twofold - firstly, it ensures code standards are met before each commit. Secondly, it provides an easy command to run all relevant linters etc. at any point during the development process.

yes, this would be amazing. I am also already using pre-commit in some other project and I can attest that it is amazing, so a PR is definitely welcome! 

Also I think in the config file, I think the linting/formatting tools could be replace with https://github.com/astral-sh/ruff
